packages/expo-video-thumbnails/ios/EXVideoThumbnails/EXVideoThumbnailsModule.m

```diff
@@ -0,0 +1,65 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+
+#import <EXVideoThumbnails/EXVideoThumbnailsModule.h>
+#import <AVFoundation/AVFoundation.h>
+#import <AVFoundation/AVAsset.h>
+#import <UIKit/UIKit.h>
+
+NSString* const OPTIONS_KEY_QUALITY = @"quality";
+NSString* const OPTIONS_KEY_TIME = "@time";
+
+@implementation EXVideoThumbnailsModule
+
+UM_EXPORT_MODULE(ExpoVideoThumbnails);
+
+- (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
+{
+    _moduleRegistry = moduleRegistry;
+    _fileSystem = [moduleRegistry getModuleImplementingProtocol:@protocol(UMFileSystemInterface)];
+}
+
+UM_EXPORT_METHOD_AS(getThumbnail,
+                    sourceFilename:(NSString *)source
+                    options:(NSDictionary *)options
+                    resolve:(UMPromiseResolveBlock)resolve
+                    reject:(UMPromiseRejectBlock)reject)
+{
+    NSURL *url = [NSURL URLWithString:source];
+    
+    if (!_fileSystem) {
+        return reject(@"E_MISSING_MODULE", @"No FileSystem module.", nil);
+    }
+    if (!([_fileSystem permissionsForURI:url] & UMFileSystemPermissionRead)) {
+        return reject(@"E_FILESYSTEM_PERMISSIONS", [NSString stringWithFormat:@"File '%@' isn't readable.", source], nil);
+    }
+
+    int timeInMs = options[OPTIONS_KEY_TIME] != nil ? [(NSNumber* )options[OPTIONS_KEY_TIME] integerValue] : 0;
```

Comment:
Objective-C is a funny language where calling methods on `nil` is "safe", so you could replace this line (and next line) with
```suggestion
    int timeInMs = [options[OPTIONS_KEY_TIME] integerValue] ?: 0;
```

Reply:
elvis operator ftw